go/vt/vtctl/reparentutil/reparent_sorter.go

```diff
@@ -75,6 +80,15 @@ func (rs *reparentSorter) Less(i, j int) bool {
 		return true
 	}
 
+	if len(rs.innodbBufferPool) != 0 {
+		if rs.innodbBufferPool[i] > rs.innodbBufferPool[j] {
+			return true
+		}
+		if rs.innodbBufferPool[j] > rs.innodbBufferPool[i] {
+			return false
+		}
+	}
+
 	// at this point, both have the same GTIDs
 	// so we check their promotion rules
 	jPromotionRule := PromotionRule(rs.durability, rs.tablets[j])
```

Comment:
I am a little on the fence on whether to use the promotion rule first or the innodb buffer pool value first in the sorting algorithm. I would love to hear other folks opinions on this.

Reply:
PromotionRule first. That's an explicit user choice.

Reply:
Fixed up!